src/Npgsql.NodaTime/TimestampTzHandler.cs

```diff
@@ -44,17 +45,24 @@ ZonedDateTime INpgsqlSimpleTypeHandler<ZonedDateTime>.Read(NpgsqlReadBuffer buf,
                     throw new NpgsqlSafeReadException(new NotSupportedException("Infinity values not supported for timestamp with time zone"));
                 return TimestampHandler.Decode(value).InZone(_dateTimeZoneProvider[buf.Connection.Timezone]);
             }
-            catch (TimeZoneNotFoundException) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
+            catch (DateTimeZoneNotFoundException e) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
```

Comment:
Maybe just catch Exception and do a switch inside on the type? Seems like this is starting to get a bit duplicatey...

Reply:
I've tried it a few different ways, and come to the conclusion that all options are ugly. 

Option 2 at least has the benefit of not needing the helper function, so pushed that for now, but happy to change it if you prefer something else.

### Option 1
```c#
catch (DateTimeZoneNotFoundException e) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
{
    throw new NpgsqlSafeReadException(LocalTimeThrowHelper(e));
}
catch (TimeZoneNotFoundException e) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
{
    throw new NpgsqlSafeReadException(LocalTimeThrowHelper(e));
}
```
### Option 2
```c#
catch (Exception e) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
{
    switch (e)
    {
    case DateTimeZoneNotFoundException _:
    case TimeZoneNotFoundException _:
        throw new NpgsqlSafeReadException(
            new TimeZoneNotFoundException(
                "The special PostgreSQL timezone 'localtime' is not supported when reading values of type 'timestamp with time zone'. " +
                "Please specify a real timezone in 'postgresql.conf' on the server, or set the 'PGTZ' environment variable on the client.",
                e));

    default:
        throw;
    }
}
```

### Option 3

```c#
catch (Exception e) when (string.Equals(buf.Connection.Timezone, "localtime", StringComparison.OrdinalIgnoreCase))
{
    throw e switch {
        DateTimeZoneNotFoundException _ => new NpgsqlSafeReadException(LocalTimeThrowHelper(e)),
        TimeZoneNotFoundException _ => new NpgsqlSafeReadException(LocalTimeThrowHelper(e)),
        _ => e
        };
}
```